### PR TITLE
Unit test 2

### PR DESCRIPTION
**Pull request by fatmahussein**
- Used jest framework for testing
- Created a test file for testing:
- a function for editing the task description.
- a function for updating an item's 'completed' status.
 - the "Clear all completed" function.
         
- Mocked  the HTML elements to test DOM manipulation functions.

### CODE REVIEW

File: src/modules/task.test.js
Comment (franclobo) [inline src/modules/task.test.js:1]: - [ ] There is an error with your test, kindly fix it to pass all the tests 👇🏽 

```javascript
$ npm run test

> webpack1@1.0.0 test
> jest

 FAIL  src/modules/task.test.js
  ● Test suite failed to run

    TypeError: Cannot redefine property: localStorage

      40 | }
      41 |
    > 42 | global.localStorage = new LocalStorageMock();
         |                    ^
      43 |
      44 | describe('check for add-delete operations', () => {
      45 |   test('properly remove task items', () => {

      at Object.<anonymous> (src/modules/task.test.js:42:20)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        2.445 s
Ran all test suites.
```

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @fatmahussein and @newtonah0, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Github flow ✔️
- Use Jest ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by m00h0** [comment]
Hey Dear code reviewer my project is working in my VSC and that's videos to check it 
[Unit-test-2-by-fatmahussein-·-Pull-Request-6-·-newtonah0-todo-list.webm](https://github.com/newtonah0/todo-list/assets/80107996/23f6195e-7a75-44cf-b0b1-e8ac7366c035)

**Review by OyePriscilla — APPROVED** [review]
Hi @fatmahussein and @newtonah0 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Used mock for functions that used localStorage ✔️ 
Tests are grouped using the description() method ✔️ 
All the written test passed as expected ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**